### PR TITLE
[2chain] connect everything and add tests

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

This PR connects the final pieces between on chain config and epoch manager. Also adds a smoke test and a cluster test for the upgrade.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

Added tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: consensus/consensus-types/src/executed_block.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ impl ExecutedBlock {
 
     pub fn transactions_to_commit(&self) -> Vec<Transaction> {
         // reconfiguration suffix don't execute
-        if self.quorum_cert().ends_epoch() {
+        if self.quorum_cert().certified_block().has_reconfiguration() {
```
Comment (gregnazario) [inline consensus/consensus-types/src/executed_block.rs:109]: Lol didn't you just write this in the other PR

File: types/src/on_chain_config/consensus_config.rs [hunk 2]
```diff
@@ -12,8 +12,7 @@ pub enum OnChainConsensusConfig {
 }
 
 impl OnChainConsensusConfig {
-    #[allow(dead_code)]
```
Comment (gregnazario) [inline types/src/on_chain_config/consensus_config.rs:15]: Yay!

**Review by gregnazario — APPROVED** [review]

**Comment by gregnazario** [comment]
@zekun000 have you tried your cluster test yet?

**Comment by zekun000** [comment]
@gregnazario yeah, that's where the two fixes come from

File: consensus/consensus-types/src/executed_block.rs (hunk 1 above)
Comment (zekun000) [reply consensus/consensus-types/src/executed_block.rs:109]: sadly yes I need to figure out how to cover this in test

**Comment by zekun000** [comment]
split the last fix out to https://github.com/diem/diem/pull/9119

**Review by gregnazario — APPROVED** [review]

**Pull request closed without merging**